netket/graph/graph.py

```diff
@@ -97,6 +97,19 @@ def __repr__(self):
             str(type(self)).split(".")[-1][:-2], self.n_nodes
         )
 
+    def __pow__(self, exponent):
+        if not isinstance(exponent, int):
+            raise ValueError(
+                "Exponent {} not valid: only integers are supported".format(exponent)
+            )
+
+        base = self
+        g = self
+        for i in range(1, exponent):
+            g = disjoint_union(g, base)
+
+        return g
+
```

Comment:
[Graph power](https://en.wikipedia.org/wiki/Graph_power) has a separate meaning in graph theory and, furthermore, the disjoint union implemented here is more like a sum than a product (which is also how it is typically denoted). All the notions of [graph product](https://en.wikipedia.org/wiki/Graph_product) add edges between the factor graphs.

Therefore, I think using `__pow__` is confusing here. In particular, `|disjoint_union(G, H)| = |G| + |H|` (where `|·|` denotes `n_nodes`), so we would have `|G**N| = N * |G|`.

Therefore, (and since this operation now is also unrelated to `hilbert**N`) I'd leave it out or - if we still want to have this method - rename it, for example to `G.disjoint_copies(N)`.

Reply:
Indeed.